### PR TITLE
yield from replacements

### PR DESCRIPTION
**Pull request by chris-rands**
More from PR #2455 / #2450 (those PRs are quite tricky to resolve conflicts on now, I suspect the relevant content to be put in separate PRs). This one required some manual edits, so please to check for typos

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/phenotype/__init__.py [hunk 1]
```diff
@@ -187,8 +187,7 @@ def parse(handle, format):
         else:
             raise ValueError("Unknown format '%s'" % format)
         # This imposes some overhead... wait until we drop Python 2.4 to fix it
```
Comment (peterjc) [inline Bio/phenotype/__init__.py]: You can from this comment line now.

**Comment by peterjc** [comment]
Looks good, just drop that comment line and this can me merged - thanks!

**Comment by chris-rands** [comment]
done- i will tackle the `IOError` next as a new PR, hopefully in the next few days

**Comment by peterjc** [comment]
@chris-rands beware of #2469 which also touches some exceptions (although that's currently held up with cross platform testing)

**Comment by peterjc** [comment]
Thanks Chris, Bow - that's merged now.

**Pull request merged**